packages/jest-core/src/watch.ts

```diff
@@ -171,12 +171,25 @@ export default function watch(
     }
 
     for (const pluginWithConfig of globalConfig.watchPlugins) {
-      const ThirdPartyPlugin = require(pluginWithConfig.path);
-      const plugin: WatchPlugin = new ThirdPartyPlugin({
-        config: pluginWithConfig.config,
-        stdin,
-        stdout: outputStream,
-      });
+      let plugin: WatchPlugin;
+      try {
+        const ThirdPartyPlugin = require(pluginWithConfig.path);
+        plugin = new ThirdPartyPlugin({
+          config: pluginWithConfig.config,
+          stdin,
+          stdout: outputStream,
+        });
+      } catch (err) {
+        return Promise.reject(
+          new Error(
+            `Failed to initialize watch plugin '${
+              pluginWithConfig.path
```

Comment:
I think it would also be nice to make the path bold.
And maybe wrap with double quotes `"` instead of single ones `'` (just my preference)

Reply:
Btw, have you considered adding an optional `name` when normalizing `watchPlugins`? We could show it if it's available (npm-published plugin, the usual case), which would definitely be nicer than a path (which is fine for local plugins)

Reply:
would we ever know if it's a package and not a local one? I guess we could sniff out `node_modules` in the path?

Reply:
the name would be whatever user entered in the config, be it `path/to/plugin` or `jest-plugin-typeahead/whatever`, so it's still more readable than resolved path, isn't it?

Reply:
I don't find these long-ish paths that problematic so not sure it's worth it, but I can do it if you both feel that it's necessary

Reply:
Added chalk / double quotes and hopefully fixed CI. Based on approval I guess we agree a more advanced path handling isn't super important here